Remotion/Web/Development.WebTesting/ScreenshotCreation/BrowserContentLocators/EdgeBrowserContentLocator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Threading;
+using System.Windows;
+using System.Windows.Automation;
+using JetBrains.Annotations;
+using OpenQA.Selenium;
+using Remotion.Web.Development.WebTesting.Utilities;
+
+namespace Remotion.Web.Development.WebTesting.ScreenshotCreation.BrowserContentLocators
+{
+  public class EdgeBrowserContentLocator : IBrowserContentLocator
+  {
+    private const string c_setWindowTitle =
+        "var w = window; while (w.frameElement) w = w.frameElement.ownerDocument.defaultView; var t = w.document.title; w.document.title = arguments[0]; return t;";
+
+    public EdgeBrowserContentLocator ()
+    {
+    }
+
+    /// <inheritdoc />
+    public Rectangle GetBrowserContentBounds (IWebDriver driver)
+    {
+      var windows = AutomationElement.RootElement.FindAll (
+              TreeScope.Children,
+              new AndCondition (
+                  new OrCondition (
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Window),
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Pane)),
+                  new PropertyCondition (AutomationElement.ClassNameProperty, "Chrome_WidgetWin_1")))
+          .Cast<AutomationElement>()
+          .ToArray();
+
+      if (windows.Length == 1)
+        return ResolveBoundsFromWindow (windows[0]);
+
+      if (windows.Length == 0)
+        throw new InvalidOperationException ("Could not find an Edge browser window in order to resolve the bounds of the content area.");
+
+      // If the result are ambiguous we try to find the browser by changing the window title
+      var automationElement = ResolveByChangingWindowTitle (driver, windows);
+
+      return ResolveBoundsFromWindow (automationElement);
+    }
+
+    [CanBeNull]
+    private AutomationElement ResolveByChangingWindowTitle (IWebDriver driver, IReadOnlyCollection<AutomationElement> windows)
+    {
+      var id = Guid.NewGuid().ToString();
+
+      var executor = (IJavaScriptExecutor) driver;
+      var previousTitle = JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, id);
+
+      var result = RetryUntil (
+          () => windows.SingleOrDefault (w => w.Current.Name.StartsWith (id)),
+          null,
+          3,
+          TimeSpan.FromMilliseconds (100));
+
+      JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, previousTitle);

```

Comment:
Would there be a benefit in putting the "change windowtitle back logic" in a finally block?
I personally can't think of one since it's just for the duration of the test but who knows

Reply:
No exceptions are to be expected after calling `JavaScriptExecutor.ExecuteStatement`, I see no benefit. Will leave open.

Reply:
> No exceptions are to be expected after calling

Is this something that's inherrently flawed/impossible or just a best practice we expect people to follow? Typically, a reliable cleaning up is always the first choice unless it's a) expensive (code or runtime wise) and b) reasonably likely to be irrelevant.

Just regarding the finally-block itself: if we can expect there to be no excepton, yes, we could probably skip it, again, most of the time, it's not worth it to consider the viability of not using a finally-block conpared to just doing it. Please also consider future maintainer of the code: without the finally block they are requried to a) notice there is a cleanup semantic and b) not introduce code that might throw while there is no finally block.

Reply:
`var previousTitle = JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, id);` (l.73) is the only place where an exception could happen, we're not expecting it tho. Adding a `try..finally` block calling the same method again does not seem ideal. But again, we don't expect it not to work.

Reply:
No, the idea would be to put the 2nd executeStatement into the finally block and the rest of the method after the 1st executeStatement into the try-block

Reply:
Then imo `b) reasonably likely to be irrelevant` is the case.

Reply:
Adding `try..finally` to express intent.